src/main/java/apoc/custom/CypherProcedures.java

```diff
@@ -111,11 +125,27 @@ public void asFunction(@Name("name") String name, @Name("statement") String stat
         GraphDatabaseAPI api;
         Procedures procedures;
         private final Log log;
+        private final CustomStatementRegistry self;
 
         public CustomStatementRegistry(GraphDatabaseAPI api, Log log) {
             this.api = api;
             procedures = api.getDependencyResolver().resolveDependency(Procedures.class);
             this.log = log;
+            self = this;
+        }
+
+        private boolean needsUpdate(long procedureLastUpdate, AtomicLong lastCheck, AtomicLong lastUpdate) {
+            long now = System.currentTimeMillis();
+            long lastCk = lastCheck.getAndSet(now);
+            if (now - lastCk < UPDATE_FREQ) {
```

Comment:
I think I might have gotten this wrong.
As it leads to starvation, if you always check within 60s then you will never update.
I think it needs to set `lastCheck.set(now)` when it doesn't return false. because then it really checks.